flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java

```diff
@@ -114,15 +115,33 @@ public Client getClient() {
 	public List<URL> getClasspaths(){
 		return classpathsToAttach;
 	}
-	
+
+	public ClassLoader getUserCodeClassLoader() {
+		return userCodeClassLoader;
+	}
+
+	public void setCurrentPlan(FlinkPlan plan) {
```

Comment:
Isn't `setDetachedPlan` a better name? It's only used for the detached mode and it's either set or null.